spark_log_parser/parsing_models/application_model.py

```diff
@@ -241,29 +217,39 @@ def __init__(self, eventlogpath, bucket=None, stdoutpath=None, debug=False):  #
                         )
 
                 elif event_type == "SparkListenerExecutorAdded":
-                    hosts.append(json_data["Executor Info"]["Host"])
-                    self.cores_per_executor = int(json_data["Executor Info"]["Total Cores"])
-                    self.num_executors = self.num_executors + 1
+                    executor_id = json_data["Executor ID"]
+                    executor_info = json_data["Executor Info"]
+
+                    executor = self.executors[executor_id]
+                    executor.id = executor_id
+                    executor.start_time = json_data["Timestamp"]
+                    executor.host = executor_info["Host"]
+                    executor.cores = int(executor_info["Total Cores"])
```

Comment:
Since `ExecutorModel` has a no-arg constructor now, we'll fill these fields in manually here